src/compiler/scala/tools/nsc/transform/LambdaLift.scala

```diff
@@ -503,6 +503,11 @@ abstract class LambdaLift extends InfoTransform {
             }
 
             treeCopy.ValDef(tree, mods, name, tpt1, factoryCall)
+          } else if(sym.enclClass.isSubClass(DelayedInitClass) && !mods.hasFlag(CAPTURED)) {
+            // Fields of DelayedInit bodies are initialized in a regular method, not in a static initializer,
+            // and must therefore not be final in bytecode (https://github.com/scala/bug/issues/11412)
+            sym.setFlag(MUTABLE)
```

Comment:
I think doing it here is a bit imprecise, for example it also catches local `val`s in methods in a `DelayedInit` subclass. I also observed some weird `owner`s of captured symbols here, maybe we're observing some intermediate state during this transformation?

In any case, I think it makes more sense to do the change in the `constructors` phase wehre `DelayedInit` is handled. I submitted https://github.com/scala/scala/pull/8239.